src/Bicep.Core.UnitTests/Diagnostics/LinterRuleTests/LinterRuleTestsBase.cs

```diff
@@ -33,7 +33,14 @@ protected void AssertLinterRuleDiagnostics(string ruleCode, string bicepText, in
         {
             AssertLinterRuleDiagnostics(ruleCode, bicepText, onCompileErrors, diags =>
             {
-                diags.Should().HaveCount(expectedDiagnosticCountForCode);
+                if (diags.Count() > 0)
+                {
+                    diags.Should().HaveCount(expectedDiagnosticCountForCode);
+                }
+                else
+                {
+                    diags.Should().HaveCount(expectedDiagnosticCountForCode);
```

Comment:
It's the same as the above one. What am I missing here?

Reply:
Had originally been different.  Thanks.